src/System.Linq.Expressions/src/System/Runtime/CompilerServices/ReadOnlyCollectionBuilder.cs

```diff
@@ -174,15 +199,21 @@ public void RemoveAt(int index)
         {
             get
             {
-                if (index >= _size)
+                // Following trick can reduce the range check by one comparison
+                if ((uint)index >= (uint)Count)
```

Comment:
Thanks, @AlexRadch.  I don't think it's just your misunderstanding; I don't think we've done a good job at applying it consistently and on describing what exactly it means.  We'll work to fix that.  In the meantime, I do appreciate your desire to fix these issues, and some of the other changes in this PR are still valuable and worth keeping.  Thank you for your efforts!